sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogV2Util.scala

```diff
@@ -126,10 +126,6 @@ private[sql] object CatalogV2Util {
 
         case update: UpdateColumnType =>
           replace(schema, update.fieldNames, field => {
-            if (!update.isNullable && field.nullable) {
```

Comment:
this check has been moved to https://github.com/apache/spark/pull/27110/files#diff-1d14ac233eac6f233c027dba0bdf871dR469

Reply:
The UT failure seems related to this one. Could you fix the UT failure?
> test("alterTable: update optional column to required fails")